src/java.base/share/classes/java/util/stream/Gatherer.java

```diff
@@ -0,0 +1,566 @@
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, A, R> Gatherer<T, A, R> ofSequential(
+            Supplier<A> initializer,
+            Integrator<A, T, R> integrator,
+            BiConsumer<A, Downstream<? super R>> finisher) {
+        return of(
+                initializer,
+                integrator,
+                defaultCombiner(),
+                finisher
+        );
+    }
+
+    /**
+     * Returns a new, parallelizable, and stateless {@code Gatherer} described
+     * by the given {@code integrator}.
+     *
+     * @param integrator the integrator function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, R> Gatherer<T, Void, R> of(Integrator<Void, T, R> integrator) {
+        return of(
+                defaultInitializer(),
+                integrator,
+                Gatherers.Value.DEFAULT.statelessCombiner,
+                defaultFinisher()
+        );
+    }
+
+    /**
+     * Returns a new, parallelizable, and stateless {@code Gatherer} described
+     * by the given {@code integrator} and {@code finisher}.
+     *
+     * @param integrator the integrator function for the new gatherer
+     * @param finisher the finisher function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, R> Gatherer<T, Void, R> of(
+            Integrator<Void, T, R> integrator,
+            BiConsumer<Void, Downstream<? super R>> finisher) {
+        return of(
+                defaultInitializer(),
+                integrator,
+                Gatherers.Value.DEFAULT.statelessCombiner,
+                finisher
+        );
+    }
+
+    /**
+     * Returns a new, parallelizable, {@code Gatherer} described by the given
+     * {@code initializer}, {@code integrator}, {@code combiner} and
+     * {@code finisher}.
+     *
+     * @param initializer the supplier function for the new gatherer
+     * @param integrator the integrator function for the new gatherer
+     * @param combiner the combiner function for the new gatherer
+     * @param finisher the finisher function for the new gatherer
+     * @param <T> the type of input elements for the new gatherer
+     * @param <A> the type of initializer for the new gatherer
+     * @param <R> the type of results for the new gatherer
+     * @throws NullPointerException if any argument is null
+     * @return the new {@code Gatherer}
+     */
+    static <T, A, R> Gatherer<T, A, R> of(
+            Supplier<A> initializer,
```

Comment:
wildcards are missing here too, Supplier<? extends A>, Integrator<? super A, ? super T, ? extends R> and BiConsumer<? super A, ? super Downstream<? super R>>.

Reply:
See separate discussion on PECS.